### ISSUE TITLE
Mention fields' immutability in spec

### ISSUE THREAD

**Issue by sanposhiho** [kubernetes/kubernetes#126068]
/sig storage
/kind cleanup

https://kubernetes.slack.com/archives/C09QZFCE5/p1720848771491719
I should've raised it as an issue for a better tracking.

> Hi, one general feedback.
I suppose maybe it’s not only sig-storage area’s problem, though. I saw several fields don’t mention they’re immutable while they actually are, which made reviewing hard when I did scheduler plugins PRs.
Examples I could list are VolumeBindingMode or ReclaimPolicy in StorageClass and CSINode.Spec.Drivers. I guess there are actually more.
Please consider updating those fields’ comments.

**Comment by thisisharrsh** [comment]
Absolutely! Comments for immutable fields are very helpful. We could also consider adding a flag or annotation to explicitly mark them as immutable. This would provide an extra layer of clarity and potentially allow for automated checks during code reviews.
